src/app/file-manager/file-explorer-upload/file-explorer-upload.component.ts

```diff
@@ -213,9 +213,9 @@ export class FileExplorerUploadComponent implements OnInit {
         this.selectedFiles
           .filter(file => file.data.type === Type.FILE)
           .forEach(file => {
-            this.dataFile =  new Item(UUID.UUID(),  file.data.name, file.data.updated, file.data.updated, file.data.size,
-                          '', file.data.path, result.directory, Type.UPLOAD, ItemStatus.PENDING, '',
-              '', '', result.preserveStructure, false, '');
+            this.dataFile = new Item(UUID.UUID(), file.data.name, file.data.updated, file.data.updated, file.data.size,
+              '', file.data.path, result.directory, Type.UPLOAD, ItemStatus.PENDING, '',
+              '', '', result.preserveStructure, false, '', '', '');
```

Comment:
Can there be a factory method in `Item()` that takes care of providing all the missing arguments? It looks like we have many calls to this constructor where a number of the parameters are always the same.

Reply:
I reviewed this. There just few calls to constructor and I don't see a big win in introducing the factory method.